packages/app-bridge/src/react/useCategorizedDocumentPages.ts

```diff
@@ -69,16 +83,53 @@ export const useCategorizedDocumentPages = (
 
         window.emitter.on('AppBridge:GuidelineDocumentPage:Action', handler);
         window.emitter.on('AppBridge:GuidelineDocumentPageTargets:Action', refetchIfPageExistsInMap);
+        window.emitter.on('AppBridge:GuidelineDocumentPage:MoveEvent', handleDocumentPageMoveEvent);
 
         return () => {
             window.emitter.off('AppBridge:GuidelineDocumentPage:Action', handler);
             window.emitter.off('AppBridge:GuidelineDocumentPageTargets:Action', refetchIfPageExistsInMap);
+            window.emitter.off('AppBridge:GuidelineDocumentPage:MoveEvent', handleDocumentPageMoveEvent);
         };
     }, [documentCategoryId, refetch, documentPages]);
 
     return { documentPages: Array.from(documentPages.values()), refetch, isLoading };
 };
 
+const previewDocumentPagesSort = (
+    documentPages: Map<number, DocumentPage>,
+    documentPage: DocumentPagesMoveEvent['documentPage'],
+    newPosition: DocumentPagesMoveEvent['position'],
+) => {
+    if (!documentPage.sort || !newPosition) {
+        return documentPages;
+    }
+
+    const previousPosition = documentPage.sort;
+    const documentPagesAsArray: DocumentPage[] = [...documentPages.values()];
+
+    documentPages.clear();
```

Comment:
Shouldn't we create a new map and avoid mutation here?

Reply:
We use `immer`'s `produce` function for calling all these methods, that takes care of this and prevents mutations.